File: app/src/main/java/org/deiverbum/app/data/entity/relation/LHGospelCanticleWithAntiphon.java
package org.deiverbum.app.data.entity.relation;

import androidx.room.Embedded;
import androidx.room.Relation;

import org.deiverbum.app.data.entity.LHAntiphonEntity;
import org.deiverbum.app.data.entity.LHGospelCanticleEntity;
import org.deiverbum.app.model.LHGospelCanticle;

/**
 * @author A. Cedano
 * @version 1.0
 * @since 2023.1
 */
public class LHGospelCanticleWithAntiphon {
    @Embedded
    public LHGospelCanticleEntity ce;
    @Relation(
            parentColumn = "antiphonFK",
            entityColumn = "antiphonID",
            entity = LHAntiphonEntity.class
    )
    public LHAntiphonEntity antifona;

    public String getAntifona() {
        return antifona.getAntifona();
    }

    public LHGospelCanticle getDomainModel(Integer tipo) {
        LHGospelCanticle ce=new LHGospelCanticle();
        ce.setTypeID(tipo);
        ce.setAntiphon(getAntifona());
        return ce;
    }
}

File: app/src/main/java/org/deiverbum/app/data/entity/relation/LHGospelCanticleWithAntiphon.kt
package org.deiverbum.app.data.entity.relation

import androidx.room.Embedded
import androidx.room.Relation
import org.deiverbum.app.data.entity.LHAntiphonEntity
import org.deiverbum.app.data.entity.LHGospelCanticleEntity
import org.deiverbum.app.model.LHGospelCanticle

/**
 * @author A. Cedano
 * @version 1.0
 * @since 2023.1
 */
class LHGospelCanticleWithAntiphon {
    @JvmField
    @Embedded
    var ce: LHGospelCanticleEntity? = null

    @JvmField
    @Relation(
        parentColumn = "antiphonFK",
        entityColumn = "antiphonID",
        entity = LHAntiphonEntity::class
    )
    var antifona: LHAntiphonEntity? = null
    fun getAntifona(): String {
        return antifona!!.antifona
    }

    fun getDomainModel(tipo: Int?): LHGospelCanticle {
        val ce = LHGospelCanticle()
        ce.typeID = tipo!!
        ce.antiphon = getAntifona()
        return ce
    }
}